### ISSUE TITLE
[D8][UX] Merge Special Menu Items module functionality in core (allow <nolink> and <separator> menu items).

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1840]
The module is [already ported to Backdrop](https://github.com/backdrop-contrib/special_menu_items) and is also used in backdropcms.org (to provide the special `<nolink>` path required for the "About Backdrop", "User Guide", "Developer Guide" and "Themer Guide" menu items of the Handbook menu).

This can be considered a sibling issue to #422 and is a sub-issue of #1839

**Comment by klonos** [comment]
...respective issue for D8: https://www.drupal.org/node/2698057

**Comment by klonos** [comment]
One more candidate for possible inclusion in 1.5.0 in order to "compensate" for not adding reference. Feel free to push to 1.6.0 is you think it's too much work.

**Comment by quicksketch** [comment]
> One more candidate for possible inclusion in 1.5.0 in order to "compensate" for not adding reference.

Heh, we'll see I guess. :wink:

The contrib module works but is sort of a hack. It modifies `theme_link()` to output... span tags. I don't think that's quite appropriate. We should make new separate theme functions for non-links like this and then build the renderable menu to use this new type (e.g. `$item['#type'] = 'link_separator'` or whatever).

It's a good feature but one we should implement cleanly.

**Comment by klonos** [comment]
Fair enough. We should not rush things into core. Especially "hacks". If we can't make it for 1.5.0, we'll push for next release and get it done right too.

**Comment by jenlampton** [comment]
I'm also not sure this is an 80% use-case. un-assigning from 1.5, let's discus at a later date :)

**Comment by klonos** [comment]
> I'm also not sure this is an 80% use-case

To be fair, the usage stats for the D7 module suggest the same, but this is targeted for inclusion in D8. Feature parity.

**Comment by jenlampton** [comment]
Let's leave the issue then, and when/if we get a contrib module we can watch it's usage statistics and assign a milestone when necessary.

**Comment by wesruv** [comment]
Working on BackdropCMS.org I found a nasty bug this introduces with our Dropdown menu.

If you add a `<nolink>` element with children you're not able to get to the children. Since the dropdown functionality is 3rd party code it probably opens up a huge can o' worms.
![image](https://cloud.githubusercontent.com/assets/5607236/20735705/dbc5f96a-b66f-11e6-88b8-e3c71a14e1e1.png)

**Comment by jenlampton** [comment]
Can you open an issue for the contrib module that adds the <nolink> entries? That's not part of core, and probs up to contrib to figure out what they should do in a drop-down menu context.

Here, for us, we can probably just remove those nolinks. They made sense in a drawer, but not so much in a menu.

**Comment by klonos** [comment]
...worth noting that the D8 implementation had an issue where links with no path were getting active class on front page. See: https://www.drupal.org/node/2838351
